api/lib/stores/global-store.js

```diff
@@ -2,67 +2,50 @@ const Sentry = require('@sentry/node')
 
 class GlobalStore {
   constructor(database) {
-    this.stores = []
     this.db = database
+    this.store = {}
     this.globalValidators = {}
+    this.validatorsLookup = {}
 
     this.dataReady = new Promise((resolve) => {
       this.resolveReady = resolve
     })
     this.dataReady.then(() => {
       console.log(`Global Store is ready`)
     })
-    this.getStores()
+    this.getReady()
   }
 
-  async getStores() {
-    const networks = await this.db.getNetworks()
-    if (networks.length !== this.stores.length) {
-      setTimeout(async () => await this.getStores(), 1000)
+  async getReady() {
+    if (Object.keys(this.store).length === 0) {
+      setTimeout(async () => await this.getReady(), 1000)
     } else {
       await this.getGlobalValidators()
       this.resolveReady()
     }
   }
 
   upsertStoreToGlobalStore(newStore) {
-    // first check if it is already there In that case update store
-    if (
-      this.stores.length > 0 &&
-      this.stores.find((store) => store.network.id === newStore.network.id)
-    ) {
-      this.stores = this.stores.reduce((updatedGlobalStores, store) => {
-        if (store.network.id === newStore.network.id) {
-          updatedGlobalStores.push(newStore)
-          return updatedGlobalStores
-        } else {
-          updatedGlobalStores.push(store)
-          return updatedGlobalStores
-        }
-      }, [])
-      // otherwise push to the list
-    } else {
-      this.stores.push(newStore)
-    }
+    this.store = newStore
+  }
+
+  createGlobalValidatorsLookup(premiumValidators) {
```

Comment:
remove

Reply:
ok, true, this we are going to just hardcode it :+1:

Reply:
for now. and put it into the db later.

Reply:
I found it just more straight-forward to just add it to DB directly